### PR TITLE
EREGCSC-1058 Skip reg parts that have already been imported

### PR DESCRIPTION
**Pull request by knollfear**
THis Pr creates a public endpoint that lists all of the dates and parts that are currently in the database.  The ECFR parser pulls in this file and if a date/part combo has been pulled in already it skips enqueuing that part.  This will lead to the parser running in only a few seconds when there is no new data to pull.

### CODE REVIEW

File: regcore/urls.py [hunk 1]
```diff
@@ -18,5 +19,6 @@
         path("<date>/title/<title>", EffectivePartsView.as_view()),
         path("<date>/title/<title>/part/<name>", EffectivePartView.as_view()),
         path("<date>/title/<title>/part/<name>/toc", EffectivePartTocView.as_view()),
+        path("title/<title>/existing", ExistingPartsView.as_view())
```
Comment (cgodwin1) [inline regcore/urls.py:22]: Recommend comma after this line for cleaner future diffs

File: regcore/views.py [hunk 2]
```diff
@@ -32,6 +33,15 @@ class Meta:
         }
 
 
+class ExistingPartSerializer(serializers.BaseSerializer):
+
+    def to_representation(self, instance):
+        return {
+            'date': instance.get("date"),
+            'partName': instance.get("partName").split(",")
```
Comment (cgodwin1) [inline regcore/views.py:41]: Same as previous

File: tools/ecfr-parser/eregs/eregs.go [hunk 3]
```diff
@@ -70,3 +78,55 @@ func PostPart(ctx context.Context, p *Part) error {
 	log.Trace("[eregs] Posted ", length, " bytes for part ", p.Name, " version ", p.Date, " in ", time.Since(start))
 	return nil
 }
+
```
Comment (cgodwin1) [inline tools/ecfr-parser/eregs/eregs.go:81]: Noticing some strange spacing errors in this file, on lines 85-101, 104-122, 130-131

File: tools/ecfr-parser/main.go [hunk 4]
```diff
@@ -136,6 +136,10 @@ func run() error {
```
Comment (cgodwin1) [inline tools/ecfr-parser/main.go:136]: More spacing issues in this file as well

**Review by cgodwin1 — COMMENTED** [review]
Looks great! Just some minor formatting issues

File: regcore/urls.py (hunk 1 above)
Comment (knollfear) [reply regcore/urls.py]: added

File: regcore/views.py (hunk 2 above)
Comment (knollfear) [reply regcore/views.py]: added

File: tools/ecfr-parser/eregs/eregs.go (hunk 3 above)
Comment (knollfear) [reply tools/ecfr-parser/eregs/eregs.go:81]: I think I improved it.  If you have any suggestions, let's sync up and talk them out.

File: tools/ecfr-parser/main.go (hunk 4 above)
Comment (knollfear) [reply tools/ecfr-parser/main.go]: same as above.

**Review by cgodwin1 — APPROVED** [review]
LGTM

**Review by theabrad — APPROVED** [review]
Pulled it down. Ran with empty DB and then ran again and saw skipped versioned parts.

**Pull request merged**